client/client-multi/client-java-desktop/src/main/java/org/kaaproject/kaa/client/logging/DesktopSQLiteDBLogStorage.java

```diff
@@ -0,0 +1,379 @@
+
+    @Override
+    public void notifyUploadFailed(int bucketId) {
+        synchronized (connection) {
+            LOG.trace("Notifying upload fail for bucket id: {}", bucketId);
+            if (resetBucketIdStatement == null) {
+                try {
+                    resetBucketIdStatement = connection.prepareStatement(PersistentLogStorageStorageInfo.KAA_RESET_BY_BUCKET_ID);
+                } catch (SQLException e) {
+                    LOG.error("Can't create bucket id reset statement", e);
+                    throw new RuntimeException(e);
+                }
+            }
+
+            try {
+                resetBucketIdStatement.setInt(1, bucketId);
+                int affectedRows = resetBucketIdStatement.executeUpdate();
+                if (affectedRows > 0) {
+                    LOG.info("Total {} log records reset for bucket id: [{}]", affectedRows, bucketId);
+                } else {
+                    LOG.warn("No log records for bucket with id: [{}]", bucketId);
+                }
+
+                long previouslyConsumedSize = consumedMemoryStorage.get(bucketId);
+                consumedMemoryStorage.remove(bucketId);
+                consumedSize += previouslyConsumedSize;
+            } catch (SQLException e) {
+                LOG.error("Failed to reset bucket with id [{}]", bucketId, e);
+            }
+        }
+    }
+
+    @Override
+    public long getConsumedVolume() {
+        return consumedSize;
+    }
+
+    @Override
+    public long getRecordCount() {
+        return recordCount;
+    }
+
+    private void initTable() throws SQLException {
+        Statement statement = null;
+        try {
+            statement = connection.createStatement();
+            statement.executeUpdate(PersistentLogStorageStorageInfo.KAA_CREATE_LOG_TABLE);
+            statement.executeUpdate(PersistentLogStorageStorageInfo.KAA_CREATE_BUCKET_ID_INDEX);
+        } finally {
+            tryCloseStatement(statement);
+        }
+    }
+
+    private void retrieveConsumedSizeAndVolume() throws SQLException {
+        synchronized (connection) {
+            Statement statement = null;
+            ResultSet rs = null;
+            try {
+                statement = connection.createStatement();
+                rs = statement.executeQuery(PersistentLogStorageStorageInfo.KAA_HOW_MANY_LOGS_IN_DB);
+                if (rs.next()) {
+                    recordCount = rs.getLong(1);
+                    consumedSize = rs.getLong(2);
+                    LOG.trace("Retrieved record count: {}, consumed size: {}", recordCount, consumedSize);
+                } else {
+                    LOG.error("Unable to retrieve consumed size and volume");
+                    throw new RuntimeException("Unable to retrieve consumed size and volume");
+                }
+            } finally {
+                tryCloseResultSet(rs);
+                tryCloseStatement(statement);
+            }
+        }
+    }
+
+    public void close() {
```

Comment:
Method's explanation is provided in LogStorage interface. Please propose more suitable name.